File: ZegoUIKitPrebuiltLiveAudioRoomDemoOC/ZegoUIKitPrebuiltLiveAudioRoomDemoOC/ViewController.m
//
//  ViewController.m
//  ZegoUIKitPrebuiltLiveAudioRoomDemoOC
//
//  Created by zego on 2024/5/28.
//

#import "ViewController.h"
@import ZegoUIKit;
@import ZegoUIKitPrebuiltLiveAudioRoom;

@interface ViewController ()
@property (nonatomic, assign) unsigned int appID;
@property (nonatomic, copy) NSString *appSign;
@property (nonatomic, copy) NSString *userID;
@property (nonatomic, copy) NSString *userName;
@property (nonatomic, copy) NSString *roomID;
@property (weak, nonatomic) IBOutlet UITextField *roomIDTextField;

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view.
  self.view.backgroundColor = [UIColor whiteColor];
  self.userID = [NSString stringWithFormat:@"%u",arc4random() % 999999];
  self.userName = @"ios_test";
  self.roomID = [NSString stringWithFormat:@"%u",arc4random() % 999999];
  self.appID = <#appID#>;
  self.appSign = <#appSign#>;
  self.roomIDTextField.text = [NSString stringWithFormat:@"%u",arc4random() % 999999];
}
- (IBAction)startLiveAudio:(id)sender {
  if(self.roomIDTextField.text.length >0){
    self.roomID = self.roomIDTextField.text;
  }
  
  ZegoUIKitPrebuiltLiveAudioRoomConfig *config = [ZegoUIKitPrebuiltLiveAudioRoomConfig host];
  ZegoUIKitPrebuiltLiveAudioRoomVC *vc = [[ZegoUIKitPrebuiltLiveAudioRoomVC alloc] init:self.appID
                                                                                appSign:self.appSign
                                                                                 userID:self.userID
                                                                               userName:self.userName
                                                                                 roomID:self.roomID
                                                                                 config:config];
//  [vc setBackgroundView:self.view];
  vc.modalPresentationStyle = UIModalPresentationFullScreen;
  [self presentViewController:vc animated:YES completion:^{
    
  }];
}

- (IBAction)watchLiveAudio:(id)sender {
  if(self.roomIDTextField.text.length >0){
    self.roomID = self.roomIDTextField.text;
  }
  ZegoUIKitPrebuiltLiveAudioRoomConfig *config = [ZegoUIKitPrebuiltLiveAudioRoomConfig audience];
  ZegoUIKitPrebuiltLiveAudioRoomVC *vc = [[ZegoUIKitPrebuiltLiveAudioRoomVC alloc] init:self.appID
                                                                                appSign:self.appSign
                                                                                 userID:self.userID
                                                                               userName:self.userName
                                                                                 roomID:self.roomID
                                                                                 config:config];
  [vc setBackgroundView:self.view];
  vc.modalPresentationStyle = UIModalPresentationFullScreen;
  [self presentViewController:vc animated:YES completion:^{
    
  }];
}


@end

File: ZegoUIKitPrebuiltLiveAudioRoomDemo/ZegoUIKitPrebuiltLiveAudioRoomDemo/ViewController.swift
//
//  ViewController.swift
//  ZegoUIKitPrebuiltLiveAudioRoomDemo
//
//  Created by zego on 2022/12/12.
//

import UIKit
import ZegoUIKitPrebuiltLiveAudioRoom

class ViewController: UIViewController {
    
    let appID: UInt32 = <#YourAppID#>
    let appSign: String = <#YourAppSign#>
    var userID: String = String(format: "%d", arc4random() % 999999)
    var userName: String = String(format: "n_%d", arc4random() % 999999)
    var roomID: String = "99999"

    @IBAction func startLiveAudio(_ sender: Any) {
        let config: ZegoUIKitPrebuiltLiveAudioRoomConfig = ZegoUIKitPrebuiltLiveAudioRoomConfig.host()
        let liveAudioVC: ZegoUIKitPrebuiltLiveAudioRoomVC = ZegoUIKitPrebuiltLiveAudioRoomVC(appID, appSign: appSign, userID: userID, userName: userName, roomID: roomID, config: config)
        liveAudioVC.modalPresentationStyle = .fullScreen
        self.present(liveAudioVC, animated: true, completion: nil)
    }
    
    @IBAction func joinLiveAudio(_ sender: Any) {
        let config: ZegoUIKitPrebuiltLiveAudioRoomConfig = ZegoUIKitPrebuiltLiveAudioRoomConfig.audience()
        let liveAudioVC: ZegoUIKitPrebuiltLiveAudioRoomVC = ZegoUIKitPrebuiltLiveAudioRoomVC(appID, appSign: appSign, userID: userID, userName: userName, roomID: roomID, config: config)
        liveAudioVC.modalPresentationStyle = .fullScreen
        self.present(liveAudioVC, animated: true)
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }

    

}

